stores/secret_store.go

```diff
@@ -0,0 +1,75 @@
+package stores
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/odpf/meteor/secrets"
+	"gocloud.dev/blob"
+	_ "gocloud.dev/blob/fileblob"
+	_ "gocloud.dev/blob/memblob"
+	"gocloud.dev/gcerrors"
+)
+
+type secretStore struct {
+	bucket *blob.Bucket
+}
+
+func NewSecretStore(storageURL string) (*secretStore, error) {
+	bucket, err := blob.OpenBucket(context.Background(), storageURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return &secretStore{
+		bucket: bucket,
+	}, nil
+}
+
+func (s *secretStore) Find(name string) (secret secrets.Secret, err error) {
+	fileName := s.buildFileName(name)
+	r, err := s.bucket.NewReader(context.Background(), fileName, nil)
+	if err != nil {
+		if s.isBlobNotFoundError(err) {
+			return secret, secrets.NotFoundError{SecretName: name}
+		}
+		return secret, err
+	}
+	defer r.Close()
+
+	err = json.NewDecoder(r).Decode(&secret)
+	if err != nil {
+		return secret, err
+	}
+
+	return
+}
+
+func (s *secretStore) Upsert(secret secrets.Secret) (err error) {
```

Comment:
Can we encrypt the secrets before storing them? Similar with when we are only storing user's encrypted password for the privacy concern.